Karl/README.md

```diff
@@ -0,0 +1,265 @@
+# Karl
+
+*Your service should be up and running. Up and running, Karl!*
+
+Karl is Current's monitoring, alerting, and continous integration service.
+
+
+## Motivation
+
+Monitoring and alerting is a data problem. Current's mission is to solve data problems easily and efficiently. And we need monitoring as we grow.
+
+And our love for stong typing and metaprogramming powers of C++ scales well to fleet configuration.
+
+On top of it, monitoring and continuous integration are too tightly coupled to be separated.
+
+## Example
+
+### Client: Claire
+
+Claire is a binary monitored by Karl.
+
+```cpp
+// Registers a `/current` HTTP endpoint on the specified port. It responds with a JSON containing:
+// * `uptime_ms`:         The uptime of the binary.
+// * `state`:             "StartingUp", "Running", or "TearingDown" <=> HTTP 500, 200 and 404 codes.
+// * `state_uptime_ms`:   The time the binary has been running in this state, in millis.
+// * `local_time`:        To ensure NTP is up and running locally.
+// * `binary_build_time`: To track how up to date the presently running version is.
```

Comment:
Thought of it a lot. Dropped for two reasons: 1) extra code to write, 2) a potential security hole.